kalite/distributed/static/js/distributed/exercises/views.js

```diff
@@ -771,23 +154,29 @@ window.ExerciseView = Backbone.View.extend({
         }
 
 
-        $(Exercises).bind("checkAnswer", this.check_answer);
+        this.listenTo(Exercises, "checkAnswer", this.check_answer);
 
-        $(Exercises).bind("gotoNextProblem", this.goto_next_problem);
+        this.listenTo(Exercises, "gotoNextProblem", this.goto_next_problem);
 
         // TODO (rtibbles): Make this nice, not horrible.
-        $(Exercises).bind("newProblem", function (ev, data) {
+        this.listenTo(Exercises, "newProblem", function (ev, data) {
             if (data.answerType=="number"||data.answerType=="decimal"||data.answerType=="rational"||data.answerType=="improper"||data.answerType=="mixed"){
-                self.software_keyboard_view = new SoftwareKeyboardView({
-                    el: self.$("#solutionarea")
-                });
+                if (typeof self.software_keyboard_view === "undefined") {
```

Comment:
Dynamically hide and show the software keyboard depending on the rendered answer type. Important for questions that have mixed answer types depending on assessment item.